### PR TITLE
Made a more accurate comment on Email sender methods

### PR DESCRIPTION
**Pull request by Richard-Strittmatter**

### CODE REVIEW

**Comment by saeideng** [comment]
according to other methods , simple and short text is enough
thank you

**Comment by ravage84** [comment]
@saeideng I do not agree with your reasoning, thus I reaopen. While I could agree that the wording isn't optimal, I still think this PR has its value.

Actually, @Richard-Strittmatter opened this PR on my behalf. I recently stumpled upon this method, wasn't sure if it made the correct thing I wanted to achieve and still wasn't sure if it's the correct one after reading the abysmal description. I had to look up the code, which is a testament to the poor quality of the current state of this method's description.

**Comment by ADmad** [comment]
@ravage84 I would suggest keeping the description short and just link to the relevant RFC page which explains the purpose of "Sender" header for emails.

**Comment by Richard-Strittmatter** [comment]
@ADmad what do you think of my amendments?

File: src/Mailer/Message.php [hunk 1]
```diff
@@ -347,23 +347,25 @@ public function getFrom(): array
...
     /**
-     * Sets "sender" address.
+     * Sets the "sender" address (acting "on behalf of")
      *
      * @param string|array $email String with email,
      *   Array with email as key, name as value or email as value (without name)
      * @param string|null $name Name
      * @return $this
      * @throws \InvalidArgumentException
+     * @link https://en.wikipedia.org/wiki/Email#Header_fields
```
Comment (ADmad) [inline src/Mailer/Message.php:357]: I would suggest a more *authentic* source https://tools.ietf.org/html/rfc2822.html#section-3.6.2

File: src/Mailer/Message.php [hunk 2]
```diff
@@ -347,23 +347,25 @@ public function getFrom(): array
     }
 
     /**
-     * Sets "sender" address.
+     * Sets the "sender" address (acting "on behalf of")
```
Comment (othercorey) [inline src/Mailer/Message.php:350]:
```suggestion
     * Sets the "sender" address (sender acting "on behalf of" from)
```

File: src/Mailer/Message.php [hunk 3]
```diff
@@ -347,23 +347,25 @@ public function getFrom(): array
...
+     * @link https://tools.ietf.org/html/rfc2822.html#section-3.6.2
      */
     public function setSender($email, ?string $name = null)
     {
         return $this->setEmailSingle('sender', $email, $name, 'Sender requires only 1 email address.');
     }
 
     /**
-     * Gets "sender" address.
+     * Gets the "sender" address (acting "on behalf of")
```
Comment (othercorey) [inline src/Mailer/Message.php:365]:
```suggestion
     * Gets the "sender" address (sender acting "on behalf of" from)
```

File: src/Mailer/Message.php (hunk 2 above)
Comment (othercorey) [reply src/Mailer/Message.php:350]: I don't know the cleanest way to write that.

Comment (ADmad) [reply src/Mailer/Message.php:350]: Which is why it's best to leave it short `Sets the "sender" address` and let users read the RFC to figure out what it does :slightly_smiling_face:

Comment (othercorey) [reply src/Mailer/Message.php:350]: agreed.

**Comment by othercorey** [comment]
@Richard-Strittmatter Sorry for the invalid tag. The change was well-made, but without a description it was caught up in our spam checks that day. We don't mark every PR we don't merge as invalid.

File: src/Mailer/Message.php (hunk 2 above)
Comment (ravage84) [reply src/Mailer/Message.php]: > That's why it's best to leave it short `Sets the "sender" address` and let users read the RFC to figure out what it does slightly_smiling_face

I still don't agree. Now I might not need to check the code, but get redirected into reading a hightly technical RFC.

I don't expect the documentation of a framework code to educate me about a techical term but to tell me if a given function will do what I think it does (judging by its name and documentation).

By removing the ``"on behalf of"`` you successfully removed the last additional info (aparr from the mentioned technical RFC) this PR could have provided... :man_shrugging:

**Pull request merged**